### PR TITLE
Project is Namespace

### PR DESCRIPTION
**Pull request by derekwaynecarr**
The PR introduces the following changes:

1. A project is now a virtual resource mapped to a Kubernetes ```Namespace```.
2. A project display name is stored as an annotation to the Namespace.

Existence of a Project/Namespace prior to it's usage will be done in a separate PR.

There are some tweaks needed to upstream fake_namespaces.go in order to make better tests, but this still provides sufficient test coverage.

Enforcing creation of the ```Project``` prior to sticking content in its namespace will be done in a subsequent PR once the Admission Control stuff in Origin is settled.

/cc @smarterclayton @deads2k @jwforres

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
I also updated the cli to now show a Namespace column when rendering a Project, since a Project has no Namespace anymore.

**Comment by smarterclayton** [comment]
To "not"?

----- Original Message -----

**Comment by derekwaynecarr** [comment]
To not, sorry.

----- Original Message -----

**Comment by derekwaynecarr** [comment]
Sent note to mailing list notifying users of planned change.

**Comment by jwforres** [comment]
So the returned JSON for the virtual project resource will have displayName on an annotation instead of directly on the object correct?  If so the UI code needs updating as well.

**Comment by derekwaynecarr** [comment]
The display name appears in Project.DisplayName like it did previously.  It also appears in Project.Annotations["displayName"] in this PR as well.

**Comment by jwforres** [comment]
Are we deprecating it from being directly on displayName?

**Comment by derekwaynecarr** [comment]
No

**Comment by smarterclayton** [comment]
Don't use camel case, I'm not sure I can win that argument with brian.

----- Original Message -----

**Comment by derekwaynecarr** [comment]
@smarterclayton - I forgot that battle was still ongoing, will update this PR.

**Comment by jwforres** [comment]
Remove the change to bindata.go, shouldn't be necessary

**Comment by derekwaynecarr** [comment]
@jwforres - I removed the change to bindata.go, we need to figure out why its reporting as needed on my local machine longer term or standardize its build in a container.

@smarterclayton - moved to use "displayname" as annotation.

any other objections/concerns?

**Comment by derekwaynecarr** [comment]
Random travis flake in integration tests in unrelated area.

File: pkg/project/registry/project/rest.go [hunk 1]
```diff
@@ -15,12 +16,12 @@ import (
 
 // REST implements the RESTStorage interface in terms of an Registry.
 type REST struct {
-	registry Registry
+	client kclient.NamespaceInterface
```
Comment (smarterclayton) [inline pkg/project/registry/project/rest.go]: Another reason this has to be this way is that we have to run on kube directly - we should make a note somewhere that it's illegal to reference a kube registry from an openshift storage object.

**Comment by smarterclayton** [comment]
Can you add an integration test for this?

**Comment by derekwaynecarr** [comment]
@smarterclayton - yes, will work on putting one together now.

**Comment by derekwaynecarr** [comment]
@smarterclayton - added integration test to commit.

**Comment by smarterclayton** [comment]
LGTM [merge]

**Pull request merged**